mars/filesystem.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 1999-2018 Alibaba Group Holding Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from gzip import GzipFile
+
+from pyarrow import LocalFileSystem as ArrowLocalFileSystem
+try:
+    import lz4, lz4.frame
+except ImportError:
+    lz4 = None
+
+from .compat import urlparse
+
+
+compressions = {
+    'gzip': lambda f: GzipFile(fileobj=f)
+}
+
+if lz4:
+    compressions['lz4'] = lz4.frame.open
+
+
+class LocalFileSystem(ArrowLocalFileSystem):
+    def __init__(self, **_):
+        super(LocalFileSystem, self).__init__()
+
+    def stat(self, path):
+        os_stat = os.stat(path)
+        stat = dict(name=path, size=os_stat.st_size, created=os_stat.st_ctime)
+        if os.path.isfile(path):
+            stat['type'] = 'file'
+        elif os.path.isdir(path):
+            stat['type'] = 'directory'
+        else:
+            stat['type'] = 'other'
+        return stat
+
+
+file_systems = {
+    'file': LocalFileSystem
+}
+
+
+def get_fs(path, storage_options):
```

Comment:
HDFS filesystem should be added, test required as well.

Reply:
As HDFS tests need extra CI settings, I will add tests in next PR.